### PR TITLE
Introduce nova Mitaka related changes

### PR DESCRIPTION
**Pull request by evrardjp**
This commit:

- Adds the new RPC-O defaults for the nova API DB
- Removes the neutron override to talk with nova
- Documents Live migration behaviour change

Connects rcbops/u-suk-dev#238

Signed-off-by: Jean-Philippe Evrard <jean-philippe.evrard@rackspace.co.uk>

### CODE REVIEW

**Comment by prometheanfire** [comment]
a better commit title would be nice (specifying nova)

File: releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+prelude: >
```
Comment (prometheanfire) [inline releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml]: not sure both prelude and other are needed for the new db

File: releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    This version introduces nova's new database,
+    nova api db.
+security:
```
Comment (prometheanfire) [inline releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml]: I do think this should be in security, but the note about speed should be here too instead of below in the other section

File: rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml [hunk 3]
```diff
@@ -97,10 +103,6 @@ swift_ceilometer_enabled: False
 keystone_ceilometer_enabled: False
 tempest_service_available_ceilometer: False
 
-neutron_neutron_conf_overrides:
-  nova:
-    endpoint_type: internal
```
Comment (prometheanfire) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: not to sure about removing this, why was this removed?

**Comment by hughsaunders** [comment]
👍

**Comment by prometheanfire** [comment]
?

File: releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml (hunk 1 above)
Comment (evrardjp) [inline releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml]: That's the usual openstack way of writing release notes... Because text will be rendered "merged", the different sections will be splitted from the prelude. The prelude explain what has been happening in general, and the other explain the more non-obvious details.

In our case, we could ask ourselves if the prelude is relevant. I think it's important for our teams to know a new db was introduced, and to be clear and upfront about it.

File: releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml (hunk 2 above)
Comment (evrardjp) [inline releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml]: I'm not sure about what you mean: You think the nova api db introduction should be in security?
What about the speed? We now are more secure. Unless the deployer explicitly decides to do so (then the information in other could be useful). I can maybe merge the note about the live migration in the security - and mentioning it would be insecure.

File: rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml (hunk 3 above)
Comment (evrardjp) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: This was part of the analysis.
Nova/neutron now communicates through internals endpoint by default, so this shouldn't appear. This could have been spotted in neutron or in nova. 

First come first served.

**Comment by evrardjp** [comment]
@prometheanfire Edited the commit message and the release note text. Would that be fine for you?

**Comment by Apsu** [comment]
RPC-AIO-Liberty-PR: Fail

File: releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml (hunk 1 above)
Comment (prometheanfire) [inline releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml]: ya, forgot about the gathering aspect of it, that's fine then

File: releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml (hunk 2 above)
Comment (prometheanfire) [inline releasenotes/notes/nova-13.0-0c2497128e677cb2.yaml]: I was talking about the switch to qemu+ssh, I'm not sure what you mean by 'mentioning it would be insecure' though...

File: rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml (hunk 3 above)
Comment (prometheanfire) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: ah, fair enough

**Comment by prometheanfire** [comment]
ya, doc update and I'd be happy

**Comment by evrardjp** [comment]
recheck_swift

File: rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml [hunk 4]
```diff
@@ -30,16 +33,19 @@ neutron_api_workers: 64
 neutron_rpc_conn_pool_size: "{{ rpc_conn_pool_size }}"
 neutron_rpc_response_timeout: "{{ rpc_response_timeout }}"
 neutron_rpc_thread_pool_size: "{{ rpc_thread_pool_size }}"
-neutron_db_max_overflow: 60
-neutron_db_max_pool_size: 120
-neutron_db_pool_timeout: 60
+neutron_db_max_overflow: "{{ db_max_overflow }}"
+neutron_db_max_pool_size: "{{ db_max_pool_size }}"
+neutron_db_pool_timeout: "{{ db_pool_timeout }}"
```
Comment (mancdaz) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: I'd say either we should also change the keystone db variables here to use the global values, or we should do both the neutron ones and the keystone ones in their own respective pull requests. Here we have half and half.

Comment (evrardjp) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: Agreed. Amending the commit right now.

**Comment by mancdaz** [comment]
👍  pending checks

**Comment by Apsu** [comment]
RPC-AIO-Liberty-PR: Fail

**Comment by evrardjp** [comment]
recheck_swift

**Comment by alextricity25** [comment]
:+1: The changes seem pretty straightforward to me, and the checks are passing. I'm going to merge this given @mancdaz review and mine.
